week_5_batch_processing/setup/macos.md

```diff
@@ -1,2 +1,52 @@
 
-Contributions are welcome!
\ No newline at end of file
+## MacOs
+
+Here we'll show you how to install Spark 3.2.1 for MacOs.
+We tested it on MacOs Monterey 12.0.1, but it should work
+for other MacOs versions as well
+
+### Installing Java
+
+Ensure Brew and Java installed in your system.
+
+`xcode-select –install`
+
+`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`
+
+`brew cask install java`
+
+
+### Installing and Setting Up Spark
+
+1. Install Scala
+
+`brew install scala@2.11`
+
+2. Install Apache Spark
+
+`brew install apache-spark`
+
+3. Add Environment Variables: 
+Add the following environment variables to your `.bash_profile` or `.zshrc`
```

Comment:
I'm not very familiar with bash on macOs, but shouldn't it be `.bashrc`? On Linux it's a more conventional way of updating the PATH variable

Reply:
I am also using zsh, but by google search I have read that ~/.bashrc is used more for a non-interactive shells and ~/.bash_profile is used for user-based operations. therefore I am going to leave as it is